### PR TITLE
Remove reserved types

### PR DESCRIPTION
**Pull request by Baccata**
Remove the out-dated type-reservation list. The only things that are reserved are language keywords now. 

Closes https://github.com/disneystreaming/smithy4s/issues/1051

### CODE REVIEW

**Comment by Baccata** [comment]
@miguel-vila, @kubukoz  although my fix works and is actually much better in the grand scheme of things (no more reserved types), it has the annoying side effect of drastically impacting the UX for AWS clients, has AWS has the annoying tendency of redefining primitive types in all its specs : 

https://github.com/aws/aws-sdk-js-v3/blob/main/codegen/sdk-codegen/aws-models/dynamodb.json#L10134-L10136 

So I think we need to take action within this same unit of work to avoid this problem by means of a ProjectionTransformer that'd turn all simple shapes that are coming from aws-namespaces into references to the "standard" shapes (as long as the shapes we're transforming don't have any traits, and have the same names/shape-type as their targets). This ProjectionTransformer would be applied by default. 

@miguel-vila, if you want to pick this, it'd be greatly appreciated

**Comment by daddykotex** [comment]
> @miguel-vila, @kubukoz although my fix works and is actually much better in the grand scheme of things (no more reserved types), it has the annoying side effect of drastically impacting the UX for AWS clients, has AWS has the annoying tendency of redefining primitive types in all its specs :
> 
> https://github.com/aws/aws-sdk-js-v3/blob/main/codegen/sdk-codegen/aws-models/dynamodb.json#L10134-L10136
> 
> So I think we need to take action within this same unit of work to avoid this problem by means of a ProjectionTransformer that'd turn all simple shapes that are coming from aws-namespaces into references to the "standard" shapes (as long as the shapes we're transforming don't have any traits, and have the same names/shape-type as their targets). This ProjectionTransformer would be applied by default.
> 
> @miguel-vila, if you want to pick this, it'd be greatly appreciated

Do you think they have something similar in-house? If they use those specs to generate their SDK, that'd mean, for example, that the Dynamo client works with dynamo.String rather than java.lang.String?

**Comment by Baccata** [comment]
@daddykotex they don't have newtypes in house, so they always render the underlying type, which solves the problem (which obviously we don't want to do, because we have newtypes)

**Comment by kubukoz** [comment]
@Baccata so the transformation would:

- replace occurrences of said types (no traits, unqualified name matches the underlying type) with the underlying type
- remove said types from the model

and it would be included as a default `smithy4sTransformations` entry? or would you make it a default in another way?

File: modules/codegen/src/smithy4s/codegen/internals/Renderer.scala [hunk 1]
```diff
@@ -658,13 +658,13 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
...
+      line"override def getMessage(): $string_ = ${field.name}"
     case field if field.tpe.isResolved =>
-      line"override def getMessage(): String = ${field.name}.orNull"
+      line"override def getMessage(): $string_  = ${field.name}.orNull"
     case field if field.required =>
-      line"override def getMessage(): String = ${field.name}.value"
+      line"override def getMessage(): $string_  = ${field.name}.value"
     case field =>
-      line"override def getMessage(): String = ${field.name}.map(_.value).orNull"
+      line"override def getMessage(): $string_  = ${field.name}.map(_.value).orNull"
```
Comment (kubukoz) [inline modules/codegen/src/smithy4s/codegen/internals/Renderer.scala]: **issue:** too many spaces (causing unnecessary changes)

**Comment by Baccata** [comment]
> replace occurrences of said types (no traits, unqualified name matches the underlying type) with the underlying type
remove said types from the model

absolutely 

> and it would be included as a default smithy4sTransformations entry? or would you make it a default in another way?

yes

**Comment by kubukoz** [comment]
I believe @denisrosca from my team will look into this next week. @baccata given this is 0.17, I suppose you want to target **this** branch with the transformation PR, so that #1052 doesn't get released without that?

**Comment by Baccata** [comment]
> I suppose you want to target this branch with the transformation PR, so that https://github.com/disneystreaming/smithy4s/pull/1052 doesn't get released without that? 

Indeed I do

**Comment by denisrosca** [comment]
@kubukoz @Baccata 

I opened https://github.com/disneystreaming/smithy4s/pull/1110 to remove AWS newtypes during code generation.

File: modules/codegen/src/smithy4s/codegen/internals/Renderer.scala (hunk 1 above)
Comment (Baccata) [reply modules/codegen/src/smithy4s/codegen/internals/Renderer.scala]: [a67eb61](https://github.com/disneystreaming/smithy4s/pull/1052/commits/a67eb61cc5c3567bc3646a1390fd1d17734aa8f1)
